doesn't build on windows, seems to be conflicting versions of the windows crate.

first error of many:
```
error[E0308]: mismatched types
   --> [...]\.cargo\registry\src\index.crates.io-1949cf8c6b5b557f\wgpu-hal-28.0.0\src\dx12\suballocation.rs:83:71
    |
 83 | ...ice: gpu_allocator::d3d12::ID3D12DeviceVersion::Device(raw.clone()),        
    |         ------------------------------------------------- ^^^^^^^^^^^ expected `ID3D12Device`, found `Direct3D12::ID3D12Device`
    |         |
    |         arguments to this enum variant are incorrect
    |
note: there are multiple different versions of crate `windows` in the dependency graph
```

`cargo tree -i windows@0.58.0`:
```
windows v0.58.0
└── gpu-allocator v0.28.0
    └── wgpu-hal v28.0.0
        ├── wgpu v28.0.0
        │   [dev-dependencies]
        │   └── naga_oil v0.21.0 (C:\Users\robfm\Documents\gamedev\lno\naga_oil)
        ├── wgpu-core v28.0.0
        │   └── wgpu v28.0.0 (*)
        └── wgpu-core-deps-windows-linux-android v28.0.0
            └── wgpu-core v28.0.0 (*)
```

`cargo tree -i windows@0.62.2`:
```
windows v0.62.2
└── wgpu-hal v28.0.0
    ├── wgpu v28.0.0
    │   [dev-dependencies]
    │   └── naga_oil v0.21.0 (C:\Users\robfm\Documents\gamedev\lno\naga_oil)
    ├── wgpu-core v28.0.0
    │   └── wgpu v28.0.0 (*)
    └── wgpu-core-deps-windows-linux-android v28.0.0
        └── wgpu-core v28.0.0 (*)
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -16,8 +16,6 @@ glsl = ["naga/glsl-in", "naga/glsl-out"]
 override_any = []
 prune = []
 allow_deprecated = []
-# forces composable modules to include "enable wgpu_ray_query"
-override_enable_ray_query = []
 
 [dependencies]
 naga = { version = "28", features = ["wgsl-in", "wgsl-out", "termcolor"] }

--- a/src/compose/mod.rs
+++ b/src/compose/mod.rs
@@ -275,6 +275,8 @@ pub struct ComposableModuleDefinition {
     modules: HashMap<ModuleKey, ComposableModule>,
     // used in spans when this module is included
     module_index: usize,
+    // any directives used in this module
+    wgsl_directives: WgslDirectives,
 }
 
 impl ComposableModuleDefinition {
@@ -585,17 +587,12 @@ impl Composer {
         language: ShaderLanguage,
         imports: &[ImportDefinition],
         shader_defs: &HashMap<String, ShaderDefValue>,
-        wgsl_directives: Option<WgslDirectives>,
+        wgsl_directives: &WgslDirectives,
     ) -> Result<IrBuildResult, ComposerError> {
         debug!("creating IR for {} with defs: {:?}", name, shader_defs);
 
-        let mut wgsl_string = String::new();
-        if let Some(wgsl_directives) = &wgsl_directives {
-            trace!("adding WGSL directives for {}", name);
-            wgsl_string = wgsl_directives.to_wgsl_string();
-        }
         let mut module_string = match language {
-            ShaderLanguage::Wgsl => wgsl_string,
+            ShaderLanguage::Wgsl => wgsl_directives.to_wgsl_string(),
             #[cfg(feature = "glsl")]
             ShaderLanguage::Glsl => String::from("#version 450\n"),
         };
@@ -853,7 +850,6 @@ impl Composer {
         demote_entrypoints: bool,
         source: &str,
         imports: Vec<ImportDefWithOffset>,
-        wgsl_directives: Option<WgslDirectives>,
     ) -> Result<ComposableModule, ComposerError> {
         let mut imports: Vec<_> = imports
             .into_iter()
@@ -987,7 +983,7 @@ impl Composer {
             module_definition.language,
             &imports,
             shader_defs,
-            wgsl_directives,
+            &module_definition.wgsl_directives,
         )?;
 
         // from here on errors need to be reported using the modified source with start_offset
@@ -1389,17 +1385,6 @@ impl Composer {
             true,
             &preprocessed_source,
             imports,
-            if cfg!(feature = "override_enable_ray_query") {
-                Some(WgslDirectives {
-                    enables: vec![EnableDirective {
-                        extensions: vec!["wgpu_ray_query".to_string()],
-                        source_location: 0,
-                    }],
-                    ..Default::default()
-                })
-            } else {
-                None
-            },
         )
         .map_err(|err| err.into())
     }
@@ -1544,6 +1529,7 @@ impl Composer {
             mut imports,
             mut effective_defs,
             cleaned_source,
+            wgsl_directives,
             ..
         } = self
             .preprocessor
@@ -1629,6 +1615,7 @@ impl Composer {
             shader_defs,
             module_index,
             modules: Default::default(),
+            wgsl_directives,
         };
 
         // invalidate dependent modules if this module already exists
@@ -1775,6 +1762,7 @@ impl Composer {
             all_imports: Default::default(),
             shader_defs: Default::default(),
             modules: Default::default(),
+            wgsl_directives,
         };
 
         let PreprocessOutput {
@@ -1801,7 +1789,6 @@ impl Composer {
                 false,
                 &preprocessed_source,
                 imports,
-                Some(wgsl_directives),
             )
             .map_err(|e| ComposerError {
                 inner: e.inner,

--- a/src/compose/test.rs
+++ b/src/compose/test.rs
@@ -1298,7 +1298,6 @@ mod test {
         output_eq!(wgsl, "tests/expected/atomics.txt");
     }
 
-    #[cfg(feature = "override_enable_ray_query")]
     #[test]
     fn test_raycasts() {
         let mut composer =

--- a/src/compose/tests/raycast/mod.wgsl
+++ b/src/compose/tests/raycast/mod.wgsl
@@ -1,3 +1,5 @@
+enable wgpu_ray_query;
+
 #define_import_path test_module
 
 @group(0) @binding(0) var tlas: acceleration_structure;

